shared/src/commonMain/kotlin/com/bunbeauty/shared/presentation/create_address/CreateAddressViewModel.kt

```diff
@@ -0,0 +1,185 @@
+package com.bunbeauty.shared.presentation.create_address
+
+import com.bunbeauty.shared.domain.asCommonStateFlow
+import com.bunbeauty.shared.domain.interactor.address.CreateAddressUseCase
+import com.bunbeauty.shared.domain.interactor.address.GetUserAddressListUseCase
+import com.bunbeauty.shared.domain.interactor.address.SaveSelectedUserAddressUseCase
+import com.bunbeauty.shared.domain.interactor.street.GetStreetsUseCase
+import com.bunbeauty.shared.presentation.SharedViewModel
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class CreateAddressViewModel(
+    private val getStreetsUseCase: GetStreetsUseCase,
+    private val createAddressUseCase: CreateAddressUseCase,
+    private val saveSelectedUserAddressUseCase: SaveSelectedUserAddressUseCase,
+) : SharedViewModel() {
+
+    private val mutableStreetListState = MutableStateFlow(CreateAddressState())
+    val streetListState = mutableStreetListState.asCommonStateFlow()
+
+    private val exceptionHandler = CoroutineExceptionHandler { coroutineContext, throwable ->
+
+        mutableStreetListState.update {
+            it.copy(
+                state = CreateAddressState.State.Error(throwable),
+                isCreateLoading = false
+            )
+        }
+    }
+
+    fun getStreetList() {
+        sharedScope.launch(exceptionHandler) {
+            val streets = getStreetsUseCase().ifEmpty {
+                mutableStreetListState.update { oldState ->
+                    oldState.copy(
+                        state = CreateAddressState.State.Empty
+                    )
+                }
+                return@launch
+            }
+
+            mutableStreetListState.update { oldState ->
+                oldState.copy(
+                    streetItemList = streets.map { street ->
+                        CreateAddressState.StreetItem(
+                            uuid = street.uuid,
+                            name = street.name,
+                        )
+                    },
+                    state = CreateAddressState.State.Success
+                )
+            }
+        }
+    }
+
+    private fun isCorrect(streetText: String): Boolean {
```

Comment:
Add more details to function name, for example `isStreetCorrect`

Reply:
+